gnome-initial-setup/pages/summary/gis-summary-page.c

```diff
@@ -236,6 +239,117 @@ done_cb (GtkButton *button, GisSummaryPage *page)
     }
 }
 
+static void
+initial_contact_app_signal (GDBusProxy *proxy,
+                            gchar *sender,
+                            gchar *signal_name,
+                            GVariant *parameters,
+                            gpointer user_data)
+{
+  GisSummaryPage *page = user_data;
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  const gchar *object, *property;
+  g_autoptr(GVariant) value;
+
+  if (g_strcmp0 (signal_name, "ObjectNotify") != 0)
+    return;
+
+  g_variant_get (parameters, "(&s&sv)", &object, &property, &value);
+
+  if (g_strcmp0 (object, "view.JSContext.globalParameters") != 0)
+    return;
+
+  if (g_strcmp0 (property, "unlocked") != 0)
+    return;
+
+  g_return_if_fail (g_variant_is_of_type (value, G_VARIANT_TYPE_BOOLEAN));
+
+  if (g_variant_get_boolean (value))
+    {
+      g_signal_handler_disconnect (priv->clippy_proxy,
+                                   priv->clippy_proxy_signal_id);
+      finish_fbe (page);
+    }
+}
+
+static void
+initial_contact_connect_to_clippy (GisSummaryPage *page)
+{
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+
+  g_dbus_proxy_call_sync (priv->clippy_proxy,
+                          "Connect",
+                          g_variant_new ("(sss)",
+                                         "view.JSContext.globalParameters",
+                                         "notify",
+                                         "unlocked"),
+                          G_DBUS_CALL_FLAGS_NONE,
+                          G_MAXINT,
+                          NULL,
+                          &error);
+
+  if (error != NULL)
+    {
+      g_critical ("Could not call Connect on DBus proxy for HackUnlock App: %s", error->message);
+
+      finish_fbe (page);
+      return;
+    }
+
+  priv->clippy_proxy_signal_id =
+    g_signal_connect (priv->clippy_proxy, "g-signal",
```

Comment:
I was surprised to learn that the D-Bus signal name is not available as the detail, but I guess that's because details get interned with GQuark so this would cause unbounded memory use.